jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/Content.java

```diff
@@ -398,6 +399,20 @@ default boolean rewind()
      */
     public interface Sink
     {
+        /**
+         * <p>Wraps the given content sink with a buffering sink.</p>
+         *
+         * @param sink the sink to write to
+         * @param bufferPool the {@link org.eclipse.jetty.io.ByteBufferPool} to use
+         * @param direct true to use direct buffers, false to use heap buffers
+         * @param maxBufferSize the maximum size of the buffer
+         * @return a Sink that writes to the given content sink
+         */
+        static Sink asBuffered(Sink sink, ByteBufferPool bufferPool, boolean direct, int maxBufferSize)
+        {
+            return new BufferedContentSink(sink, bufferPool, direct, maxBufferSize);
+        }
+
         /**
          * <p>Wraps the given content sink with an {@link OutputStream}.</p>
```

Comment:
I also think we need to do something with asOutputStream, specifically to avoid the double write when someone writes the last content and then closes.
Wrapping the output stream as a BufferedOutputStream does not help.

I think we perhaps need a
```java
        static OutputStream asOutputStream(Sink sink, ByteBufferPool bufferPool, boolean direct, int maxBufferSize)
```

The default implementation using a small buffer, even without a pool.
We might also need a static on `Response` that takes both the request and response and uses the requests to get the buffer pool etc.

Reply:
Since `Sink`s can be chained, doing:
`OutputStream os = Content.Sink.asOutputStream(Content.Sink.asBuffered(response, bufferPool, direct, maxSize));`
or more simply:
`OutputStream os = Content.Sink.asOutputStream(Response.asBufferedSink(request, response));`
would do, wouldn't it?
I've added a `Response.asBufferedOutputStream(Request, Response)` static helper as a shortcut to the above:
`OutputStream os = Response.asBufferedOutputStream(request, response);`